addon/index.js

```diff
@@ -63,6 +66,7 @@ export class QueryParamAwarePretender extends Pretender {
   };
 
   normalizeURLs = false;
+  reasonForNoMatching = null;
```

Comment:
I think this is a great idea!

Reply:
Changed by adding an attribute to the request object.

Reply:
Is there some advantage here to putting a property on the request instead of using the request as a key in a weakmap?


```js
let requestRejectionReasons = new WeakMap();

// ...

requestRejectionReasons.set(request, reason)
```

The advantage of weakmaps here is to avoid polluting a public object with what is essentially a private field, so it helps to avoid users relying on something that isn't intended to be public API